toolkit/tools/imagegen/diskutils/diskutils.go

```diff
@@ -433,7 +423,7 @@ func WaitForDevicesToSettle() error {
 }
 
 // CreatePartitions creates partitions on the specified disk according to the disk config
-func CreatePartitions(diskDevPath string, disk configuration.Disk, rootEncryption configuration.RootEncryption,
+func CreatePartitions(targetOs targetos.TargetOs, diskDevPath string, disk configuration.Disk, rootEncryption configuration.RootEncryption,
```

Comment:
An opinion here, since this is a new repo, there is a chance that as time goes on, we might end up adding more and more options here.
How about we have a struct with default values, and on top of that struct we implement a method `CreateParititions`.
This might help us in only modifying the code at a single place where we have the initial struct and the rest of the code might not need any modifications :)
Just an opinion

Reply:
Most of the options for this function are included in the `Disk` struct. So, it is probably unlikely that this function will expand over time. Also, this function is only called in 4 places. So, IMHO, this function is probably ok as it is.